sql: remove calls to CreateTestServerParams

yuzefovich opened the pull request:
This PR removes calls to `tests.CreateTestServerParams` outside of `sql_test` tests as well as does some miscellaneous cleanups along the way. The rationale for doing this is that vast majority of callers didn't actually need the setup performed by the helper but inherited the disabling of test tenant. It seems more prudent for each test to be explicit about the kind of testing knobs and settings it needs.

Informs: #76378.
Epic: CRDB-18499.

Release note: None

yuzefovich committed:
sql: remove calls to CreateTestServerParams

This commit removes all calls to `tests.CreateTestServerParams` outside
of `sql` and `sql_test` tests. The following commit will move and
unexport this method.

Miscellaneous items:
- enable ccl license in many touched packages
- adjust some tests to work with test tenant
- rename pkg/sql/pgrepl/pgrepl_test.go -> pkg/sql/pgrepl/main_test.go
- unexport `AOSTClause` from `sqlstats.TestingKnobs`
- sprinkle some log scopes.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/107765)
<!-- Reviewable:end -->

yuzefovich committed:
catalog: run TestValidateSystemSchemaAfterBootStrap against test tenant

This commit refactors `TestValidateSystemSchemaAfterBootStrap` to be run
against the test tenant. Given that the system schema is different for
system and application tenants, a new `bootstrap_tenant` file is
introduced.

Release note: None

yuzefovich committed:
sql/tests: remove CreateTestTenantParams helper

It doesn't seem that useful - it only sets SQLStats testing knob which
I don't think any of the tests actually need.

Release note: None

yuzefovich committed:
sql/tests: move and unexport createTestServerParams

Now that `tests.CreateTestServerParams` is only used from within `sql`
folder, this helper is now moved into the `sql` package. Also, in order to
unexport the helper to prevent it from polluting other packages
unnecessarily, this commit removes all callers of the helper that are in
`sql` test package (as opposed to `sql_test`).

Release note: None

knz reviewed (approved):
LGTM with nits.

> Reviewed 5 of 5 files at r1, 72 of 72 files at r2, 8 of 8 files at r3, 45 of 46 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107765#-:-Narihmc7rR7s8XLGunP:b3590pt)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @stevendanna and @yuzefovich)

___
*[`pkg/sql/schemachanger/scbuild/builder_test.go` line 296 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/107765#-NarhkrR2jrKkZ11ZRGD:-NarhkrR2jrKkZ11ZRGE:bn9srbz) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb6935fb75369896c6656d432816d12312aab92d/pkg/sql/schemachanger/scbuild/builder_test.go#L296)):*
> ```Go
> 
> 	systemDB := s.SystemLayer().SQLConn(t, "")
> 	_, err := systemDB.Exec(`ALTER TENANT ALL SET CLUSTER SETTING "spanconfig.tenant_limit" = 10000000`)
> ```

May I suggest a setting override? The ALTER TENANT ALL propagates async with a rangefeed.
___
*[`pkg/sql/tests/random_schema_test.go` line 46 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/107765#-Nari4ce9gDo3GeAyr5S:-Nari4ce9gDo3GeAyr5T:b-7ropln) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb6935fb75369896c6656d432816d12312aab92d/pkg/sql/tests/random_schema_test.go#L46)):*
> ```Go
> 
> 	systemDB := s.SystemLayer().SQLConn(t, "")
> 	_, err := systemDB.Exec(`ALTER TENANT ALL SET CLUSTER SETTING "spanconfig.tenant_limit" = 10000000`)
> ```

ditto


<!-- Sent from Reviewable.io -->

yuzefovich reviewed (commented):
TFTR!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/107765#-:-Nas4gghCOpFXgtJMm2M:b-oduykh)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @knz and @stevendanna)

___
*[`pkg/sql/schemachanger/scbuild/builder_test.go` line 296 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/107765#-NarhkrR2jrKkZ11ZRGD:-Nas4Oyk4rtIgLgTNRQl:b-ypcxjv) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb6935fb75369896c6656d432816d12312aab92d/pkg/sql/schemachanger/scbuild/builder_test.go#L296)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

May I suggest a setting override? The ALTER TENANT ALL propagates async with a rangefeed.
</blockquote></details>

The setting is not exported, so I reverted back to the testing knob.
___
*[`pkg/sql/tests/random_schema_test.go` line 46 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/107765#-Nari4ce9gDo3GeAyr5S:-Nas4f9-0j9VSXLOdBn8:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/bb6935fb75369896c6656d432816d12312aab92d/pkg/sql/tests/random_schema_test.go#L46)):*
<details><summary><i>Previously, knz (Raphael 'kena' Poss) wrote…</i></summary><blockquote>

ditto
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->
